processor/src/operations/io_ops.rs

```diff
@@ -157,7 +160,14 @@ impl Process {
         let value = self.stack.get(1);
 
         // write the value to the memory and get the previous word
-        let mut old_word = self.chiplets.write_mem_element(ctx, addr, value)?;
+        let mut old_word = {
+            let old_word = self.chiplets.memory.get_old_value(ctx, addr);
+            let new_word = [value, old_word[1], old_word[2], old_word[3]];
```

Comment:
I think this logic is from before we merged element-addressable memory. So just a heads up that this will need to be updated